diff --git a/xs/makefiles/lin/xst.mk b/xs/makefiles/lin/xst.mk
index a9190972..45f6e6f5 100644
--- a/xs/makefiles/lin/xst.mk
+++ b/xs/makefiles/lin/xst.mk
@@ -43,7 +43,6 @@ C_OPTIONS = \
 	-fno-common \
 	-DINCLUDE_XSPLATFORM \
 	-DXSPLATFORM=\"xst.h\" \
-	-DmxDebug=1 \
 	-DmxNoConsole=1 \
 	-DmxParse=1 \
 	-DmxRun=1 \
@@ -65,9 +64,9 @@ else
 	C_OPTIONS += -O3
 endif
 
-LIBRARIES = -ldl -lm -lpthread -latomic
+# LIBRARIES = -ldl -lm -lpthread -latomic
 
-LINK_OPTIONS = -rdynamic
+# LINK_OPTIONS = -rdynamic
 
 OBJECTS = \
 	$(TMP_DIR)/xsAll.o \
diff --git a/xs/sources/xsCommon.c b/xs/sources/xsCommon.c
index 2eb2c5d5..87099d98 100644
--- a/xs/sources/xsCommon.c
+++ b/xs/sources/xsCommon.c
@@ -526,7 +526,7 @@ char* fxCStackLimit()
     	void* stackAddr = pthread_get_stackaddr_np(self);
    		size_t stackSize = pthread_get_stacksize_np(self);
 		return (char*)stackAddr - stackSize + (8 * 1024);
-	#elif mxLinux
+	#elif mxLinuxXX
 		pthread_attr_t attrs;
 		if (pthread_getattr_np(pthread_self(), &attrs) == 0) {
     		void* stackAddr;
